src/artifactContentRenderers/artifactContentRendering.ts

```diff
@@ -0,0 +1,47 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ */
+
+import * as vscode from "vscode";
+import * as sarif from "sarif";
+import { BinaryArtifactContentRenderer } from "./binaryArtifactContentRenderer";
+import { TextArtifactContentRenderer } from "./textArtifactContentRenderer";
+import { RenderingArtifactContentRenderer } from "./renderingArtifactContentRenderer";
+
+export interface ArtifactContentRenderer {
+    /**
+     * Computes a range from the content it will render.
+     * @param region The region for the artifact content.
+     */
+    rangeFromRegion?(region: sarif.Region): vscode.Range | undefined;
+
+    render(artifactUri: vscode.Uri): string;
+
+    specificUriExtension?: string;
```

Comment:
Also unused.

Reply:
It is used to create the URI for VSCode to detect the language. It is referenced in the location factory as well as the artifact content file system provider.

I forgot to comment these functions, so I've done so.

---
In reply to: [411732865](https://github.com/microsoft/sarif-vscode-extension/pull/252#discussion_r411732865) [](ancestors = 411732865)